dataactbroker/handlers/fileHandler.py

```diff
@@ -479,9 +498,134 @@ def checkD1File(self, submission_id=None):
 
         return JsonResponse.create(StatusCode.OK, response)
 
+    def loadGenerateRequest(self):
+        """ Pull information out of request object and return it
+
+        Returns: tuple of submission ID and file type
+
+        """
+        requestDict = RequestDictionary(self.request)
+        if not (requestDict.exists("submission_id") and requestDict.exists("file_type")):
+            exc = ResponseException("Generate file route requires submission_id and file_type",
+                                    StatusCode.CLIENT_ERROR)
+            return JsonResponse.error(exc, exc.status)
+
+        submission_id = requestDict.getValue("submission_id")
+        file_type = requestDict.getValue("file_type")
+        return submission_id, file_type
+
+    def generateFile(self):
+        """ Start a file generation job for the specified file type """
+        submission_id, file_type = self.loadGenerateRequest()
+        job = self.interfaces.jobDb.getJobBySubmissionFileTypeAndJobType(submission_id, self.EXTERNAL_FILE_TYPE_MAP[file_type], "file_upload")
+        # Check prerequisites on upload job
+        if not self.interfaces.jobDb.runChecks(job.job_id):
+            exc = ResponseException("Must wait for completion of prerequisite validation job", StatusCode.CLIENT_ERROR)
+            return JsonResponse.error(exc, exc.status)
+        if file_type == "D1":
+            # TODO could this function return S3 file location
+            result = self.generateD1File()
+        elif file_type == "D2":
+            # TODO could this function return S3 file location
+            result = self.generateD2File()
+        elif file_type == "E":
+            #TODO call file E generation function
+            pass
+        elif file_type == "F":
+            #TODO call file F generation function
+            pass
+        else:
+            exc = ResponseException("File type must be either D1, D2, E or F", StatusCode.CLIENT_ERROR)
+            return JsonResponse.error(exc, exc.status)
+
+        # TODO mark S3 file location in DB
+        # Mark file generation upload as finished
+        self.interfaces.jobDb.markJobStatus(job.job_id,"finished")
+        # Return same response as check generation route
+        return self.checkGeneration()
+
+    def checkGeneration(self):
```

Comment:
I could make those optional arguments here, but as this function is the first called from the /check_generate_status/ route it needs to be able to pull the values from the request if they're not provided.